pkg/kubectl/cmd/util/factory.go

```diff
@@ -354,6 +354,19 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 					return nil, errors.New("provided options object is not a PodLogOptions")
 				}
 				return c.Pods(t.Namespace).GetLogs(t.Name, opts), nil
+
+			case *api.ReplicationController:
+				opts, ok := options.(*api.PodLogOptions)
+				if !ok {
+					return nil, errors.New("provided options object is not a PodLogOptions")
+				}
+				selector := labels.SelectorFromSet(t.Spec.Selector)
+				pod, err := GetFirstPod(c, t.Namespace, selector)
```

Comment:
The original intent of this was to be a command to find "the first useful thing".  A usability case we have today is that it's unreasonably hard (in terms of number of steps) to go from a deployment to the logs of a pod.  The majority of deployments and RCs that will exist on a Kube system (probably 90%?) will have a single pod.  Maybe we want to display a warning message that says "connecting to the oldest pod xoueontuh" which would also make this still usable without being unclear.